### PR TITLE
Phoebe/calendly2

### PR DESCRIPTION
**Pull request by phoebeli23**
# ✨ New in this PR ✨

## One liner: what did you do? 🚨 

Finish? data processing for Scheduling Page

## Coverage 🙆‍♀️
_use this section to break up your task into submodules and track progress_ 
 - [x] Filter out past appointments
 - [x] Filter out cancelled appointments
 - [x] Sort appointments by date
 - [x] Refresh scheduling page when leaving and reopening tab
 - [x] Refresh scheduling page when web browser closed 

## How can the reviewer test your code? 👩‍💻
1. Make sure your client document has a "cases" collection. The buttons to schedule an appointment will only appear if you have a case of the given type (`Daca`, `CriminalRecord`, or `Citizenship`) with status `SchedApt`. See my [client document ](https://console.firebase.google.com/u/2/project/siren-6099f/firestore/data/~2Fclients~2FoFYwnvTkPzTDg5TEqKj48Zgokxl1~2Fcases) for an example. 
2. Navigate to Schedule page
3. Click one of the links to open calendly
4. Schedule an appointment with Phoebe's personal Calendly
5. Close the web browser
6. You should immediately see the appointment pop up
7. Reschedule or cancel with the email you receive from Calendly

Note: Scheduling page refreshes when you go to another tab and then return 

## Any bugs you encountered or still having trouble with? 🐛
_describe how you solved them such that someone can follow your explanation_ 

## Resources 📔
_link online resources and related PRs_ 

🧜‍♀️ cc: @shannonbonet @phoebeli23

### CODE REVIEW

File: src/database/queries.tsx [hunk 1]
```diff
@@ -233,15 +233,46 @@ export const deleteQuestion = async (question: Question) => {
...
+          activeMap.delete(a.id);
+        } else {
+          activeMap.set(a.id, a);
+        }
+      }
+    });
+
+    // return upcoming appointments sorted by start time
+    return Array.from(activeMap.values()).sort((a, b) => {
+      return a.startTime < b.startTime ? -1 : 1;
```
Comment (shannonbonet) [inline src/database/queries.tsx:274]: Wooahh curious - how does this -1 : 1 sort things by start time?

File: src/screens/Schedule/ScheduleScreen.tsx [hunk 2]
```diff
@@ -38,20 +43,28 @@ const ScheduleScreen = () => {
...
+  }, [isFocused, detectBrowserClose]);
 
   const openCalendlyInBrowser = async (link: string) => {
-    await WebBrowser.openBrowserAsync(link);
+    if (Platform.OS === 'ios') {
+      await WebBrowser.openBrowserAsync(link);
+      setDetectBrowserClose(!detectBrowserClose);
+      console.log('ios');
+    } else {
+      await WebBrowser.openAuthSessionAsync(link, '');
```
Comment (shannonbonet) [inline src/screens/Schedule/ScheduleScreen.tsx:59]: Is there a reason for handling browser differently for Android vs iOS? This seems to work fine for iOS, but when I try opening the browser in Android emulator, I get this error. 
Screen Shot 2021-11-21 at 5 42 12 PM
ee

**Comment by shannonbonet** [comment]
This looks good! Some functionality comments: 

- If I reschedule my appointment, I won't get an updated event email to cancel/reschedule again. This means the event never gets deleted from the client's upcoming. I think this can be fixed through Calendly settings to send emails for reschedules?

TO DO (not in this PR) 
- Create a case collection upon client intake form

File: src/database/queries.tsx (hunk 1 above)
Comment (phoebeli23) [reply src/database/queries.tsx:274]: sort() takes a comparator function 
- if the return value of the func is <0, that means a < b
- if the rv is >0, a > b

File: src/screens/Schedule/ScheduleScreen.tsx (hunk 2 above)
Comment (phoebeli23) [reply src/screens/Schedule/ScheduleScreen.tsx:59]: `openBrowserAsync` behaves differently on ios and android. for ios, it behaves as desired, returning from the function call when the browser is closed. on android, it returns when the browser is opened, so we need to use a different function to detect when the web browser is closed.

Comment (phoebeli23) [reply src/screens/Schedule/ScheduleScreen.tsx:59]: hmm it runs correctly for me - did you npm install?

**Pull request merged**